pkg/chaosdaemon/server.go

```diff
@@ -30,45 +35,96 @@ var log = ctrl.Log.WithName("chaos-daemon-server")
 
 //go:generate protoc -I pb pb/chaosdaemon.proto --go_out=plugins=grpc:pb
 
-// Server represents an HTTP server for tc daemon
-type Server struct {
+// Config contains the basic chaos daemon configuration.
+type Config struct {
+	HTTPPort int
+	GRPCPort int
+	Host     string
+	Runtime  string
+}
+
+// Server represents a grpc server for tc daemon
+type daemonServer struct {
 	crClient ContainerRuntimeInfoClient
 }
 
-func newServer(containerRuntime string) (*Server, error) {
+func newDaemonServer(containerRuntime string) (*daemonServer, error) {
 	crClient, err := CreateContainerRuntimeInfoClient(containerRuntime)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Server{
+	return &daemonServer{
 		crClient: crClient,
 	}, nil
 }
 
-// StartServer starts chaos-daemon.
-func StartServer(host string, port int, containerRuntime string) error {
-	lis, err := net.Listen("tcp", fmt.Sprintf("%s:%d", host, port))
+func newGRPCServer(containerRuntime string, reg *prometheus.Registry) (*grpc.Server, error) {
+	ds, err := newDaemonServer(containerRuntime)
 	if err != nil {
-		log.Error(err, "failed to listen")
-		return err
+		return nil, err
 	}
 
-	s := grpc.NewServer(grpc.UnaryInterceptor(utils.TimeoutServerInterceptor))
-	chaosDaemonServer, err := newServer(containerRuntime)
-	if err != nil {
-		log.Error(err, "failed to create server")
-		return err
+	grpcMetrics := grpc_prometheus.NewServerMetrics()
+	grpcMetrics.EnableHandlingTimeHistogram(
+		grpc_prometheus.WithHistogramBuckets([]float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20}),
```

Comment:
I will reduce the buckets. Seems like the latency cannot come to 20s, it won't be that slow

Reply:
Updated